src/overlays/actors/ovl_En_Bomjima/z_en_bomjima.c

```diff
@@ -45,83 +83,1070 @@ const ActorInit En_Bomjima_InitVars = {
+                    this->unk_2C0 = 0;
+                    if (Rand_ZeroOne() < 0.5f) {
+                        func_80BFE494(this, 19, 1.0f);
+                    } else {
+                        func_80BFE494(this, 0, 1.0f);
+                    }
+                    this->unk_2A2 = 0;
+                    this->unk_2D0 = 0.0f;
+                    break;
+                }
+            }
+
+            x = this->unk_2A4.x - this->actor.world.pos.x;
+            z = this->unk_2A4.z - this->actor.world.pos.z;
+
+            if ((this->unk_2BE == 0) || (sqrtf(SQ(x) + SQ(z)) < 4.0f)) {
+                this->unk_2C0 = Rand_S16Offset(20, 20);
+                if (!(this->unk_2C0 & 1)) {
+                    func_80BFE494(this, 19, 1.0f);
+                } else {
+                    func_80BFE494(this, 0, 1.0f);
+                }
+                this->unk_2A2 = 0;
+                this->unk_2D0 = 0.0f;
+            } else if (sp84 >= 0.0f) {
+                Math_ApproachF(&this->actor.world.pos.x, this->unk_2A4.x, 0.3f, this->unk_2D0);
+                Math_ApproachF(&this->actor.world.pos.z, this->unk_2A4.z, 0.3f, this->unk_2D0);
+                Math_ApproachF(&this->unk_2D0, 1.0f, 0.3f, 0.5f);
+            }
+            break;
+    }
+}
+
+void func_80BFEA94(EnBomjima* this, GlobalContext* globalCtx) {
+    Actor* actor = globalCtx->actorCtx.actorList[ACTORCAT_PROP].first;
+
+    while (actor != NULL) {
+        if (actor->id != ACTOR_EN_BOMBAL) {
+            actor = actor->next;
+            continue;
+        }
+
+        this->unk_2F0 = (EnBombal*)actor;
+        Math_Vec3f_Copy(&this->unk_2B0, &actor->world.pos);
+        if (this->unk_2F4 == 0) {
+            this->unk_2F4 = this->unk_2F0->actor.cutscene;
+        }
+        func_80BFEB1C(this);
+        break;
+    }
+}
+
+void func_80BFEB1C(EnBomjima* this) {
+    func_80BFE494(this, 1, 1.0f);
+    func_80BFE65C(this);
+    this->unk_2A0 = 0;
+    this->actionFunc = func_80BFEB64;
+}
+
+void func_80BFEB64(EnBomjima* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s32 pad[2];
+    Vec3f sp40;
+
+    func_80BFE32C(this, globalCtx, 0);
+    if (player->transformation == PLAYER_FORM_DEKU) {
+        if (gSaveContext.weekEventReg[73] & 0x20) {
+            this->unk_2C8 = 3;
+            func_80BFE32C(this, globalCtx, 3);
+        } else if (gSaveContext.weekEventReg[77] & 2) {
+            this->unk_2C8 = 11;
+            func_80BFE32C(this, globalCtx, 2);
+        }
+    } else if (player->transformation == PLAYER_FORM_HUMAN) {
+        if (gSaveContext.weekEventReg[84] & 0x80) {
+            this->unk_2C8 = 0;
+            func_80BFE32C(this, globalCtx, 3);
+        } else if (gSaveContext.weekEventReg[85] & 1) {
+            this->unk_2C8 = 11;
+            func_80BFE32C(this, globalCtx, 2);
+        }
+    }
+
+    if (Text_GetFaceReaction(globalCtx, 0x11)) {
```

Comment:
```suggestion
    if (Text_GetFaceReaction(globalCtx, 0x11) != 0) {
```
This function returns a textId, which isn't boolean